x-pack/platform/packages/shared/onechat/onechat-server/src/events.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { OnechatEvent } from '@kbn/onechat-common';
+import type { RunContextStackEntry } from './runner';
+
+export enum OnechatRunEventType {
+  toolCall = 'toolCall',
+  toolResponse = 'toolResponse',
+}
+
+/**
+ * Base set of meta for all onechat run events.
+ */
+export interface OnechatRunEventMeta {
+  /**
+   * Current runId
+   */
+  runId: string;
+  /**
+   * Execution stack
+   */
+  stack: RunContextStackEntry[];
+}
+
+/**
+ * Public-facing events, as received by the API consumer.
+ */
+export type OnechatRunEvent<
```

Comment:
so you're thinking public events are designed to be surfaced on the UI. Could you give an example of what an internal run event would be?